com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java

```diff
@@ -157,9 +159,23 @@ private void removeOldOrphanedFiles(Date expirationTime) {
     }
 
     private void removeActivitiHistoricData(Date expirationTime) {
+        Set<String> historicProcessIds = new HashSet<String>();
         activitiFacade.getHistoricProcessInstancesFinishedAndStartedBefore(expirationTime)
             .stream()
-            .forEach(historicProcessInstance -> activitiFacade.deleteHistoricProcessInstance(historicProcessInstance.getId()));
+            .forEach(historicProcessInstance -> addProcessAndSubProcesses(historicProcessIds, historicProcessInstance.getId()));

```

Comment:
this could be done without using the extra Set :)

Reply:
did it without an extra Set :)